selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/preferences/CheckErrorsWarningsTabTest.java

```diff
@@ -62,67 +61,87 @@
 
-    List<String> expectedErrorsWarningsList = readFile(errorsWarningFilePath);
-    String embedCode = readFileToString(embedCodeFilePath);
+    // open workspace
+    ide.open(workspace);
     projectExplorer.waitItem(PROJECT_NAME);
     notificationsPopupPanel.waitProgressPopupPanelClose();
 
+    // prepare file for testing
     testProjectServiceClient.updateFile(
         workspace.getId(), PATH_TO_CLASS_IN_SPRING_PACKAGE, embedCode);
 
+    // expand project explorer tree and wait LS init
     projectExplorer.quickExpandWithJavaScript();
     loader.waitOnClosed();
+    consoles.waitJDTLSProjectResolveFinishedMessage(PROJECT_NAME);
+  }
+
+  @Test
+  public void errorsWarningTest() throws Exception {
+    final String expectedTabTitle = "AppController";
+    final URL errorsWarningFilePath = getClass().getResource("errors-warnings");
+    final List<String> expectedErrorsWarningsList = readFile(errorsWarningFilePath);
+
+    // open file
     projectExplorer.openItemByVisibleNameInExplorer(expectedTabTitle + ".java");
     editor.waitTabIsPresent(expectedTabTitle);
     editor.waitTabSelection(0, expectedTabTitle);
     editor.waitActive();
 
+    // check markers default settings
     menu.runCommand(TestMenuCommandsConstants.Profile.PROFILE_MENU, PREFERENCES);
     preferences.waitPreferencesForm();
     preferences.waitMenuInCollapsedDropdown(Preferences.DropDownJavaCompilerMenu.ERRORS_WARNINGS);
     preferences.selectDroppedMenuByName(Preferences.DropDownJavaCompilerMenu.ERRORS_WARNINGS);
-    preferences.getItemsFromErrorWarningsWidget();
     assertEquals(preferences.getItemsFromErrorWarningsWidget(), expectedErrorsWarningsList);
-
     preferences.close();
     consoles.closeProcessesArea();
+
+    // change and check markers settings for displaying warnings
     menu.runCommand(TestMenuCommandsConstants.Profile.PROFILE_MENU, PREFERENCES);
     changeAllSettingsInErrorsWarningsTab(Preferences.DropDownValueForErrorWaitingWidget.WARNING);
     editor.waitAnnotationsAreNotPresent(ERROR_OVERVIEW);
-    assertTrue(editor.getMarkersQuantity(WARNING_OVERVIEW) >= 12);
-    assertEquals(editor.getMarkersQuantity(WARNING), 22);
+    waitWarningMarkersCount();
 
+    // change and check markers settings for displaying errors
     editor.waitAnnotationsAreNotPresent(ERROR_OVERVIEW);
     menu.runCommand(TestMenuCommandsConstants.Profile.PROFILE_MENU, PREFERENCES);
     changeAllSettingsInErrorsWarningsTab(Preferences.DropDownValueForErrorWaitingWidget.ERROR);
-    assertEquals(editor.getMarkersQuantity(ERROR_OVERVIEW), 12);
-    assertEquals(editor.getMarkersQuantity(ERROR), 22);
-    editor.waitAnnotationsAreNotPresent(WARNING_OVERVIEW);
-    assertTrue(editor.getMarkersQuantity(ERROR_OVERVIEW) >= 12);
-    assertEquals(editor.getMarkersQuantity(ERROR), 22);
+    waitErrorMarkersCount();
 
+    // change and check markers settings for ignoring all markers
+    editor.waitAnnotationsAreNotPresent(WARNING_OVERVIEW);
```

Comment:
is it about `markers` or `annotations`? Can we use only one word to avoid confusion?